src/compiler/checker.ts

```diff
@@ -15107,7 +15107,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 isInJSFile(declaration) &&
                 isValueSignatureDeclaration(declaration) &&
                 !hasJSDocParameterTags(declaration) &&
-                !getJSDocType(declaration);
+                !getJSDocType(declaration) &&
+                !getContextualSignatureForFunctionLikeDeclaration(declaration);

```

Comment:
Does this change the semantics of `@type`, or is it specific to `@satisfies`? For example, in

```js
/** @type {(uuid: string) => string}
const f = uuid => uuid
```

I recall that `@type` intentionally is *not* treated as a contextual signature in that case, but I could be mistaken. I believe that instead the `@type` tag is looked up and treated the same as if it were written:

```js
const f = /** @type {(uuid: string) => }*/(uuid => uuid)
```

That is, treated like a cast. Probably `@satisfies` should be the same.

Reply:
It changes the semantics of the second example but not the first one. `@type` above the declaration is treated as the contextual signature (`getContextualSignature` calls `getSignatureOfTypeTag` and returns that if it's found).

That said, I'm not sure if this is a bad change. This flag has one purpose - it changes how `getMinArgumentCount` gets computed. If the types of parameters are assigned in just any way, I think it's surprising that the resulting signature might get treated as "untyped" and that it might allow calls without all (seemingly) required parameters. Just take a look at the situation reported in [this thread](https://github.com/microsoft/TypeScript/issues/58580) and at the discussion happening there. While this PR here doesn't solve the problem described there with `@type` applied to parameters, the root cause is similar - despite the call being typed it's treated as untyped. This creates weird discrepancies between TS and JS files.

Reply:
An extra data point:
```ts
const f1 = /** @type {(uuid: string) => {}}*/(uuid => uuid)
f1() // error

const f2 = /** @satisfies {(uuid: string) => {}}*/(uuid => uuid)
f2() // ok, wat?
```

The reason behind the difference here is that with the cast the inner signature is untyped but the final type of `f1` is sourced from the casted type itself. Since that's a type node~ it can't be untyped. 

To give more context on the above:
- `checkAssertionWorker` computes `exprType` with an untyped signature (but its parameters are typed!)
- but it returns `getTypeFromTypeNode(type)`

This can't work like that with `@satisfies` as the `@satifies` merely provides some contextual typing - it doesn't "override" the expression type.

Reply:
Thanks, that explanation helps. I was getting side-tracked by having the jsdoc on the declaration instead of directly on the arrow.